features/323-login-id/type.md

```diff
@@ -0,0 +1,95 @@
+# Login ID Type
+
+## Background
+
+Skygear Auth supports login with login id and password. This document defines the types of login id.
+
+## Validation and Normalization
+
+Validation and normalization are defined base on login id type. When user signup with login ids, Skygear Auth will validate the login ids and reject the signup if it won't pass. After passing the validation, Skygear Auth will store the normalized login id for login.
+
+When user login with login id, login id key is optional. If login id key is provided, Skygear Auth will normalize the input by type, query the principal and perform the login. If login id key is not provided, Skygear Auth will walk through all login id keys, normalize the input and query the principal. If multiple principals are found, the login process will be stopped and error will be returned.
+
+## Types
+
+This section defines the login id types and its validation and normalization rules.
+
+- [Email](#Email)
+- [Username](#Username)
+- [Phone](#Phone)
+- [Raw](#Raw)
+
+### Email
+
+#### Validation
+
+- [RFC 5322 address](https://tools.ietf.org/html/rfc5322#section-3.4.1) (Compulsory)
+
+#### Normalization
+
+- Case fold and encode the domain part to punycode (Compulsory)
```

Comment:
Does the punycode conversion do the case fold already?

Reply:
I think no. Punycode conversion only convert it from unicode to ascii, so the case is not converted.